### PR TITLE
fix(developer): use var not let in definitions

### LINKED ISSUE #4716: bug(developer): LM compiler generates code with let keyword which breaks older Android devices
**Describe the bug**

The `let` keyword is available in Chrome 49+, but our baseline is 37. The LM compiler is generating code like this:

```
(function() {
'use strict';
let definitions = {
  defaults: {
    version: "undefined",
    applyCasing: function defaultApplyCasing(casing, text) {
    switch (casing) {
        case 'lower':
            return text.toLowerCase();
        case 'upper':
            return text.toUpperCase();
        case 'initial':
            let headCode = text.charCodeAt(0);
            // The length of the first code unit, as measured in code points.
            let headUnitLength = 1;
            // Is the first character a high surrogate, indicating possible use of UTF-16 
            // surrogate pairs?  Also, is the string long enough for there to BE a pair?
            if (text.length > 1 && headCode >= 0xD800 && headCode <= 0xDBFF) {
                // It's possible, so now we check for low surrogates.
                let lowSurrogateCode = text.charCodeAt(1);
                if (lowSurrogateCode >= 0xDC00 && lowSurrogateCode <= 0xDFFF) {
                    // We have a surrogate pair; this pair is the 'first' character.
                    headUnitLength++;
                }
            }
            // Capitalizes the first code unit of the string, leaving the rest intact.
            return text.substring(0, headUnitLength).toUpperCase() // head - uppercased
                .concat(text.substring(headUnitLength)); // tail - lowercased
    }
}
  },  model: {
    searchTermToKey: function defaultCasedSearchTermToKey(wordform, applyCasing) {
    // While this is a bit WET, as the basic `defaultSearchTermToKey` exists and performs some of
    // the same functions, repetition is the easiest way to allow the function to be safely compiled 
    // with ease by use of `.toString()`.
    return Array.from(wordform
        .normalize('NFKD')
        // Remove any combining diacritics (if input is in NFKD)
        .replace(/[\u0300-\u036F]/g, '')) // end of `Array.from`
        .map(c => applyCasing('lower', c))
        .join('');
}
  },
  applyCasing: function(caseToApply, text) {
        return definitions.defaults.applyCasing(caseToApply, text);
      },
  searchTermToKey: function(text) {
      return definitions.model.searchTermToKey(text, definitions.applyCasing);
    }
};
LMLayerWorker.loadModel(new models.TrieModel(
...
```

The breadcrumb from KEYMAN-ANDROID-JJ helps:

```
KMKeyboard.sendKMWError
packageID: sil_euro_latin
keyboardID: sil_euro_latin
keyboardType: INAPP
cm_message: Uncaught SyntaxError: Unexpected strict mode reserved word
cm_sourceID: file:///data/data/com.tavultesoft.kmapro.debug/app_data/models/nrc.en.mtnt/nrc.en.mtnt.model.js
keyboardName: EuroLatin (SIL) Keyboard
cm_lineNumber: 3
```

So this is triggering the error seen in #4647. A fix to the compiler and re-release of the two affected models in Keyman Cloud (nrc.en.mtnt and katelem.ann-latn.getat) should address that issue.

**To Reproduce**

1. See #4647.

---

**Keyman Developer:**
 - OS: Win 10 
 - Keyman Developer version: 14.0.264-beta

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #4716.

We still support Chrome 37 on Android 5.0, which does not permit use of the `let` keyword in normal conditions.

### CODE REVIEW

**Comment by darcywong00** [comment]
Do you have an engineering build of nrc.en.mtnt with this lm compiler for @MakaraSok to try loading in his emulator?

**Comment by mcdurdin** [comment]
> I'd be curious if the arrow function @jahorton mentions in [#4647 (comment)](https://github.com/keymanapp/keyman/issues/4647#issuecomment-800719544) is compatible.

Hmm. [caniuse.com suggests nope](https://caniuse.com/?search=%3D%3E). Will build and test here as well. Out of time tonight though.

**Comment by mcdurdin** [comment]
> Do you have an engineering build of nrc.en.mtnt with this lm compiler for @MakaraSok to try loading in his emulator?

Working on this.

**Comment by mcdurdin** [comment]
Good call on testing. I had built it but was waiting for this to build Keyman Developer. Doing the build locally is a little tedious (lots of long command lines) but uncovered a handful of extra problems.

Here's a version of the model I have built with the latest kmlmc: [nrc.en.mtnt.model.zip](https://github.com/keymanapp/keyman/files/6161836/nrc.en.mtnt.model.zip)

And I uncovered #4719 while doing this testing...

**Comment by mcdurdin** [comment]
I have tested this locally with Android 5.1.1 (Chromium 39) in an emulator and it loaded without error and worked fine.

File: developer/js/source/lexical-model-compiler/model-definitions.ts [hunk 1]
```diff
@@ -120,18 +120,18 @@ export class ModelDefinitions {
...
    * the model itself, as the model will need to refer to the definitions herein.
    */
   compileDefinitions(): string {
     let defn: string = '';
-    defn += `let ${PSEUDOCLOSURE} = {\n`
+    defn += `var ${PSEUDOCLOSURE} = {\n`
 
     // ----------------------
     // START - the 'defaults', which are common within the same Developer version.
     defn += `  defaults: {\n    version: "${this.defaults.version}"`;
```
Comment (jahorton) [inline developer/js/source/lexical-model-compiler/model-definitions.ts:133]: Not that it's important for _this_ PR, but this is the site affected by #4705.

**Review by jahorton — APPROVED** [review]

File: developer/js/source/lexical-model-compiler/model-definitions.ts (hunk 1 above)
Comment (mcdurdin) [reply developer/js/source/lexical-model-compiler/model-definitions.ts:133]: FWIW, all of these issues were introduced in #3770. For future reference if required.

**Comment by mcdurdin** [comment]
Merging now; we will need to update the two models mentioned in #4716 after we get the updated compiler.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 14.0.265-beta](https://keyman.com/downloads/releases/beta/14.0.265)

**Comment by mcdurdin** [comment]
Test builds of the packages using latest compiler on keymanapp/lexical-models#125:
* https://build.palaso.org/repository/download/Keyman_Models_TestPullRequests/261801:id/models/release/nrc/nrc.en.mtnt/build/nrc.en.mtnt.model.kmp
* https://build.palaso.org/repository/download/Keyman_Models_TestPullRequests/261801:id/models/release/katelem/katelem.ann-latn.getat/build/katelem.ann-latn.getat.model.kmp

I've tried these here and they worked on my emulator.

**Comment by MakaraSok** [comment]
https://github.com/keymanapp/keyman/issues/4647#issuecomment-803794403

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.20-alpha](https://keyman.com/downloads/releases/alpha/15.0.20)

**Pull request merged**